Build failure with gcc13

https://github.com/xmake-io/xmake-repo/actions/runs/10627760358/job/29461413865
```console
inc/lift/http.hpp:19:19: error: found ‘:’ in nested-name-specifier, expected ‘::’
   19 | enum class method : uint8_t
      |                   ^
      |                   ::
inc/lift/http.hpp:19:12: error: ‘method’ has not been declared
   19 | enum class method : uint8_t
      |            ^~~~~~
inc/lift/http.hpp:20:1: error: expected unqualified-id before ‘{’ token
   20 | {
      | ^
inc/lift/http.hpp:32:16: error: ‘method’ was not declared in this scope; did you mean ‘method_put’?
   32 | auto to_string(method m) -> const std::string&;
      |                ^~~~~~
      |                method_put
inc/lift/http.hpp:32:26: error: expected ‘,’ or ‘;’ before ‘->’ token
   32 | auto to_string(method m) -> const std::string&;
      |                          ^~
  > in src/http.cpp
```

Need to add `cstdint` header
https://gcc.gnu.org/gcc-13/porting_to.html

--- a/.githooks/readme-template.md
+++ b/.githooks/readme-template.md
@@ -39,6 +39,7 @@ ${EXAMPLE_README_CPP}
 C++17 compilers tested
     g++-9
     g++-11
+    g++-13
     clang-9
     clang-14
 CMake
@@ -54,6 +55,7 @@ stdc++fs
 Tested on:
     ubuntu:20.04
     ubuntu:22.04 (with custom libcurl built)
+    ubuntu:24.04
     fedora:31
 ```
 
@@ -170,7 +172,7 @@ Using `nginx` as the webserver with the default `fedora` configuration.
 
 File bug reports, feature requests and questions using [GitHub Issues](https://github.com/jbaldwin/liblifthttp/issues)
 
-Copyright © 2017-2023, Josh Baldwin
+Copyright © 2017-2024, Josh Baldwin
 
 [badge.language]: https://img.shields.io/badge/language-C%2B%2B17-yellow.svg
 [badge.license]: https://img.shields.io/badge/license-Apache--2.0-blue

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -149,6 +149,71 @@ jobs:
                     cd build-release-clang
                     ctest -VV
 
+    build-ubuntu-24-04:
+        name: ubuntu-24.04
+        runs-on: ubuntu-latest
+        container:
+            image: ubuntu:24.04
+            env:
+                TZ: America/New_York
+                DEBIAN_FRONTEND: noninteractive
+        services:
+            nginx:
+                image: nginx:stable
+            haproxy:
+                image: jbaldwindh/liblifthttp-automation-haproxy:0.1
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
+            -   name: apt
+                run: |
+                    apt-get update
+                    apt-get -y upgrade
+                    apt install -y build-essential software-properties-common
+                    add-apt-repository ppa:ubuntu-toolchain-r/test
+                    apt-get install -y \
+                        cmake \
+                        git \
+                        ninja-build \
+                        g++-13 \
+                        clang \
+                        zlib1g-dev \
+                        libcurl4-openssl-dev \
+                        libuv1-dev \
+                        curl
+            -   name: check-haproxy
+                run: |
+                    curl -v --user guest:guestpassword -x "http://haproxy:3128" "http://nginx/index.html"
+            -   name: build-release-g++
+                run: |
+                    mkdir build-release-g++
+                    cd build-release-g++
+                    cmake \
+                        -GNinja \
+                        -DCMAKE_BUILD_TYPE=Release \
+                        -DCMAKE_C_COMPILER=gcc-13 \
+                        -DCMAKE_CXX_COMPILER=g++-13 \
+                        ..
+                    ninja
+            -   name: build-release-clang
+                run: |
+                    mkdir build-release-clang
+                    cd build-release-clang
+                    cmake \
+                        -GNinja \
+                        -DCMAKE_BUILD_TYPE=Release \
+                        -DCMAKE_C_COMPILER=clang \
+                        -DCMAKE_CXX_COMPILER=clang++ \
+                        ..
+                    ninja
+            -   name: test-release-g++
+                run: |
+                    cd build-release-g++
+                    ctest -VV
+            -   name: test-release-clang
+                run: |
+                    cd build-release-clang
+                    ctest -VV
 
     build-fedora-31:
         name: fedora-31

--- a/LICENSE
+++ b/LICENSE
@@ -1,4 +1,4 @@
-Copyright 2017-2020 Josh Baldwin
+Copyright 2017-2024 Josh Baldwin
 
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.

--- a/README.md
+++ b/README.md
@@ -122,6 +122,7 @@ int main()
 C++17 compilers tested
     g++-9
     g++-11
+    g++-13
     clang-9
     clang-14
 CMake
@@ -137,6 +138,7 @@ stdc++fs
 Tested on:
     ubuntu:20.04
     ubuntu:22.04 (with custom libcurl built)
+    ubuntu:24.04
     fedora:31
 ```
 
@@ -253,7 +255,7 @@ Using `nginx` as the webserver with the default `fedora` configuration.
 
 File bug reports, feature requests and questions using [GitHub Issues](https://github.com/jbaldwin/liblifthttp/issues)
 
-Copyright © 2017-2023, Josh Baldwin
+Copyright © 2017-2024, Josh Baldwin
 
 [badge.language]: https://img.shields.io/badge/language-C%2B%2B17-yellow.svg
 [badge.license]: https://img.shields.io/badge/license-Apache--2.0-blue

--- a/inc/lift/http.hpp
+++ b/inc/lift/http.hpp
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <string>
+#include <cstdint>
 
 #include <curl/curl.h>
 

--- a/inc/lift/lift_status.hpp
+++ b/inc/lift/lift_status.hpp
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <string>
+#include <cstdint>
 
 namespace lift
 {

--- a/inc/lift/query_builder.hpp
+++ b/inc/lift/query_builder.hpp
@@ -4,6 +4,7 @@
 #include <string>
 #include <string_view>
 #include <vector>
+#include <cstdint>
 
 namespace lift
 {

--- a/inc/lift/resolve_host.hpp
+++ b/inc/lift/resolve_host.hpp
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <string>
+#include <cstdint>
 
 namespace lift
 {

--- a/src/executor.cpp
+++ b/src/executor.cpp
@@ -85,7 +85,7 @@ auto executor::prepare() -> void
             curl_easy_setopt(m_curl_handle, CURLOPT_POST, 1L);
             break;
         case http::method::put:
-            curl_easy_setopt(m_curl_handle, CURLOPT_PUT, 1L);
+            curl_easy_setopt(m_curl_handle, CURLOPT_UPLOAD, 1L);
             break;
         case http::method::delete_t:
             curl_easy_setopt(m_curl_handle, CURLOPT_CUSTOMREQUEST, "DELETE");

